DSharpPlus/Entities/DiscordGuild.cs

```diff
@@ -2099,6 +2099,43 @@ public Task<DiscordGuildTemplate> ModifyTemplateAsync(string code, string name =
         /// <exception cref="Exceptions.ServerErrorException">Thrown when Discord is unable to process the request.</exception>
         public Task<DiscordGuildTemplate> DeleteTemplateAsync(string code)
             => this.Discord.ApiClient.DeleteGuildTemplateAsync(this.Id, code);
+
+        /// <summary>
+        /// Gets this guild's membership screening form.
+        /// </summary>
+        /// <returns>This guild's membership screening form.</returns>
+        /// <exception cref="Exceptions.ServerErrorException">Thrown when Discord is unable to process the request.</exception>
+        public Task<DiscordGuildMembershipScreening> GetMembershipScreeningFormAsync()
+            => this.Discord.ApiClient.GetGuildMembershipScreeningFormAsync(this.Id);
+
+        /// <summary>
+        /// Modifies this guild's membership screening form.
+        /// </summary>
+        /// <param name="action">Action to perform</param>
+        /// <returns>The modified screening form.</returns>
+        /// <exception cref="Exceptions.UnauthorizedException">Thrown when the client doesn't have the <see cref="Permissions.ManageGuild"/> permission, or community is not enabled on this guild.</exception>
+        /// <exception cref="Exceptions.ServerErrorException">Thrown when Discord is unable to process the request.</exception>
+        public async Task<DiscordGuildMembershipScreening> ModifyMembershipScreeningFormAsync(Action<MembershipScreeningEditModel> action)
+        {
+            var mdl = new MembershipScreeningEditModel();
+            action(mdl);
+            if(mdl.Terms != null)
+            {
+                var terms = new DiscordGuildMembershipScreeningField()
```

Comment:
See my comment below about the screening field, they should be able to pass in their own collection of terms.